rocky/rocky/views/scheduler.py

```diff
@@ -34,13 +34,21 @@ def get_task_filters(self) -> dict[str, Any]:
             "scheduler_id": f"{self.task_type}-{self.organization.code}",
             "task_type": self.task_type,
             "plugin_id": None,  # plugin_id present and set at plugin detail
-            **self.get_form_data(),
+            **self.get_task_filter_form_data(),
         }
 
-    def get_form_data(self) -> dict[str, Any]:
+    def get_task_filter_form_data(self) -> dict[str, Any]:
         form_data = self.get_task_filter_form().data.dict()
         return {k: v for k, v in form_data.items() if v}
 
+    def count_active_task_filters(self):
+        form_data = self.get_task_filter_form_data()
+
+        for task_filter in form_data.copy():
```

Comment:
You don't need to copy an object to modify it and calculate the sum of its elements. Instead, you could count the original elements and subtract the filtered elements. Copying is usually expensive and should be avoid if not necessary. For example:

```python3
count = len(form_data)
for task_filter in form_data:
    if task_filter in ("observed_at", "ooi_id"):
        count -= 1

return count
```

Or functional:

```python3
return sum((task_filter for task_filter in form_data if task_filter not in ("observed_at", "ooi_id")))
```